rocks-bin/src/remove.rs

```diff
@@ -1,43 +1,63 @@
+use std::io;
+
 use clap::Args;
-use eyre::Result;
+use eyre::{eyre, Result};
+use itertools::Itertools;
 use rocks_lib::{
     config::{Config, LuaVersion},
-    manifest::ManifestMetadata,
-    package::{PackageName, PackageVersion, RemotePackage},
+    package::PackageReq,
     progress::{MultiProgress, Progress},
-    tree::Tree,
+    tree::{RockMatches, Tree},
 };
 
 #[derive(Args)]
 pub struct Remove {
     /// The name of the rock to remove.
-    name: PackageName,
-    /// The name of the version to remove.
-    version: Option<PackageVersion>,
+    packages: Vec<PackageReq>,
 }
 
+// TODO(vhyrro): Add `all:xyz>=2.0` JJ-like syntax.
 pub async fn remove(remove_args: Remove, config: Config) -> Result<()> {
-    let metadata = ManifestMetadata::from_config(&config).await?;
+    let tree = Tree::new(config.tree().clone(), LuaVersion::from(&config)?)?;
 
-    let target_version = remove_args
-        .version
-        .or(metadata.latest_version(&remove_args.name).cloned())
-        .unwrap();
+    let package_matches = remove_args
+        .packages
+        .iter()
+        .map(|package_req| tree.match_rocks(package_req))
+        .try_collect::<_, Vec<_>, io::Error>()?;
 
-    let tree = Tree::new(config.tree().clone(), LuaVersion::from(&config)?)?;
+    let (packages, nonexistent_packages, duplicate_packages) = package_matches.into_iter().fold(
+        (Vec::new(), Vec::new(), Vec::new()),
+        |(mut p, mut n, mut d), rock_match| {
+            match rock_match {
+                RockMatches::NotFound(req) => n.push(req),
+                RockMatches::Single(package) => p.push(package),
+                RockMatches::Many(packages) => d.extend(packages),
+            };
+
+            (p, n, d)
+        },
+    );
```

Comment:
note: I was hoping to use `partition_map` here but it only partitions into two collections :|